Automattic/jetpack
# Better handling on My Jetpack product interstitials if site already has prodcut

## Proposed changes:

* This diff improves the behavior of the product interstitial pages in My Jetpack if the site already has a product.
* Most notably, the table style interstitial now shows an "install" button and omits the price and any free offering if the site already has the product in question

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Make sure your test site does not have any boost purchases to start with
* With this PR applied, navigate to `wp-admin/admin.php?page=my-jetpack/#add-boost`
* Note that, without a boost purchase, you should see both paid and free options:
![Screenshot 2024-03-26 at 3 49 14 PM](https://github.com/Automattic/jetpack/assets/18016357/d4239fa5-a90f-4c6a-9e48-bf594eb29a46)
* Purchase a Jetpack Boost plan
* After the purchase, navigate again to `wp-admin/admin.php?page=my-jetpack/#add-boost`
* Now, the page should only show the benefits of the boost product with an "Install Jetpack Boost" CTA
![Screenshot 2024-03-26 at 3 50 29 PM](https://github.com/Automattic/jetpack/assets/18016357/7920be59-bc8d-48fb-93e3-0556b67e5233)
* Clicking the "Install Jetpack Boost" CTA should take you to the boost plugin page
